### PR TITLE
Add documentation for seed based discovery.

### PR DESCRIPTION
**Pull request by andreidan**
## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by andreidan** [comment]
> I was thinking about adding some changes entries already, but I guess we'll at the end write a fully fledged upgrade section, so lgtm.

Yeah, I thought the same about changes entries. And the upgrading section is going to be a guide section (I thought of treating that as part of the other story)

File: blackbox/docs/config/cluster.rst [hunk 1]
```diff
@@ -309,74 +309,63 @@ resources. The following settings change the behaviour of those queries.
...
-   CrateDB 1.0.3 and removed in CrateDB 1.1.
+   Contains a list of names or IP addresses of the master-eligible nodes which
+   will vote in the very first election of a cluster that's bootstrapping for
+   the first time. By default this is not set, meaning it expects this node to
+   join an already formed cluster.
+   In development mode, with no discovery settings configured, this step is
+   performed by the nodes themselves, but this auto-bootstrapping is designed
+   to aim development and is not safe for production. In production you must
+   explicitly list the names or IP addresses of the master-eligible nodes whose
+   votes should be counted in the very first election.
```
Comment (seut) [inline blackbox/docs/config/cluster.rst]: We should mention imho that this is only needed when a complete fresh cluster is started. For existing cluster full (re-)starts this is not needed.

Comment (andreidan) [reply blackbox/docs/config/cluster.rst]: This is mentioned in the first paragraph "Contains a list of names or IP addresses of the master-eligible nodes which will vote in the very first election of a cluster that's bootstrapping for the first time." and also "whose votes should be counted in the very first election." Not sure how to make it more explicit. Do you have a suggestion here?

Comment (seut) [reply blackbox/docs/config/cluster.rst]: You're right, I was reading too fast. Sounds already good, ignore my comment! ;-)

**Pull request merged**